### ISSUE TITLE
When trying to build using 0.39.0 under El Capitan and XCode 7.1, I'm getting linker errors

### ISSUE THREAD

**Issue by kcorey** [CocoaPods/CocoaPods#4455]
I'm slightly uncomfortable writing this as an 'issue', as I expect that the issue is with our Podfile, or the complexity of our build structure.

The app was originally built back in the XCode 5 era, I think.  No idea what version of Pods was in use back then.

The way our app builds looks like this:
![app structure](https://cloud.githubusercontent.com/assets/535595/10759082/3d75c8d2-7cad-11e5-83d4-a584b7150807.png)
We compile a static library with the bulk of our app, and then have 6 targets that build separate white label versions of our app.

The Pods are meant to be linked into the "CLEngine", and then the apps just link to "CLEngine".

The Podfile is shown below.

This has worked up to XCode 6.4 and cocoapods 0.37.2.  Now, uploads to iTunesconnect seem to be failing, so it's time to upgrade to XCode.

I've now upgraded to XCode 7.1. At first I tried 0.37.2, but got loads of errors like this: "CoreUI: CUICatalog: couldn't get 'kMGArtworkGraphicsFeatureSetClass' from MGCopyAnswer got".

Then I tried cocoapods 0.39.0.  Did a "rm -rf Pods Podfile.lock CLEngine.xcworkspace", and then a "pod install", which proceeds without issue.

When I tried to compile in XCode, the build proceeds all the way to the end, and the link fails as the symbols from the Pods are not found.

`ld: warning: directory not found for option '-L/Users/kencorey/Documents/CLEngine_iOS/Pods/build/Debug-iphoneos'
Undefined symbols for architecture arm64:
  "_CGImageSourceCopyPropertiesAtIndex", referenced from:
      +[UIImage(GIF) sd_frameDurationAtIndex:source:] in libCLEngine.a(UIImage+GIF.o)
  "_kCGImagePropertyGIFDictionary", referenced from:
      +[UIImage(GIF) sd_frameDurationAtIndex:source:] in libCLEngine.a(UIImage+GIF.o)
  "_kCGImagePropertyGIFUnclampedDelayTime", referenced from:
      +[UIImage(GIF) sd_frameDurationAtIndex:source:] in libCLEngine.a(UIImage+GIF.o)
  "_kCGImagePropertyGIFDelayTime", referenced from:
      +[UIImage(GIF) sd_frameDurationAtIndex:source:] in libCLEngine.a(UIImage+GIF.o)
  "_CGImageSourceCreateWithData", referenced from:
      +[UIImage(GIF) sd_animatedGIFWithData:] in libCLEngine.a(UIImage+GIF.o)
  "_CGImageSourceCreateImageAtIndex", referenced from:
      +[UIImage(GIF) sd_animatedGIFWithData:] in libCLEngine.a(UIImage+GIF.o)
  "_OBJC_CLASS_$_AFJSONRequestOperation", referenced from:
      objc-class-ref in libCLEngine.a(JSONWebService.o)
  "_OBJC_METACLASS_$_DTAttributedTextView", referenced from:
      _OBJC_METACLASS_$_TextStimulusView in libCLEngine.a(TextStimulusView.o)
  "_OBJC_CLASS_$_DTAttributedTextContentView", referenced from:
      l_OBJC_$_CATEGORY_DTAttributedTextContentView_$_CreateLinkButton in libCLEngine.a(DTAttributedTextContentView+CreateLinkButton.o)
  "_OBJC_CLASS_$_TDDatePickerController", referenced from:
      objc-class-ref in libCLEngine.a(AccountCreationViewController.o)
...
`

My Podfile is:
`
#Podfile

#workspace 'CLEngine'

#link_with "CLEngine"

#uncomment thisline to define a global platform for your project
platform :ios, "7.0"

#use_frameworks!

target "CLEngine" do
  pod 'PureLayout'
  pod 'BlocksKit'
  pod 'YLProgressBar'
  pod 'Toast', '~> 2.4'
  pod 'DateTools'
  pod 'WYPopoverController', '~> 0.3.7'
  pod 'DTCoreText', "~> 1.6.13"
  pod 'CSURITemplate', "~> 0.3"
  pod 'AFNetworking', "~> 1.1"
  pod 'AFOAuth2Client',git:"https://github.com/crowdlab-uk/AFOAuth2Client.git"
  pod 'MagicalRecord', "~> 2.3"
  pod 'TDSemiModal', "~> 1.0.0"
  pod 'OpenUDID', "~> 1.0.0"
  pod 'TPKeyboardAvoiding', "~> 1.2.4"
  pod 'APLKeyboardControls', "~> 0.1.2"
  pod 'AFAmazonS3Client',git:"https://github.com/crowdlab-uk/AFAmazonS3Client.git", branch:"1.x"
  pod 'ObjectiveSugar', "~> 1.1.0"
end

target :CLEngineTests, :exclusive => true do
  pod 'Kiwi', "> 2.3.0"
end

target :CrowdLabTests, :exclusive => true do
  link_with 'YourWordTests'
  pod 'KIF','~> 3.0'
end

require 'fileutils'
post_install do |installer|
  ['debug','release'].each { |buildtype| remove_xml_from(buildtype) }
end

FILEPATH = "Pods/Target Support Files/Pods-CLEngine"

def remove_xml_from(buildtype)
  puts "Removing xml2 from '#{FILEPATH}/Pods-CLEngine.#{buildtype}.xcconfig'"

  File.open("#{FILEPATH}/Pods-CLEngine.#{buildtype}.xcconfig", "r") do |oldfile|
    File.open("#{FILEPATH}/Pods-CLEngine.#{buildtype}.xcconfigreplace", "w") do |newfile|
      while (line = oldfile.gets)
        newfile.puts "#{line.gsub(/\-l\"xml2\"/,'')}"
      end
    end
  end

  FileUtils.mv("#{FILEPATH}/Pods-CLEngine.#{buildtype}.xcconfigreplace","#{FILEPATH}/Pods-CLEngine.#{buildtype}.xcconfig");
rescue => err
  puts "Exception: #{err}"
  err
end
`

Is the podfile wrong?  Is this a bug with cocoapods?

**Comment by segiddins** [comment]
It looks like you're changing the XCConfig for that pod... Can you share the diff between what CocoaPods generates and what you're replacing it with?

**Comment by kcorey** [comment]
The code was to remove an extra '-lxml' library from the compilation for the apps.  

The CLEngine.xcconfig file had this line:
OTHER_LDFLAGS = $(inherited) -ObjC -l"AFAmazonS3Client" -l"AFNetworking" -l"AFOAuth2Client" -l"APLKeyboardControls" -l"BlocksKit" -l"CSURITemplate" -l"DTCoreText" -l"DTFoundation" -l"DateTools" -l"MagicalRecord" -l"NSData+Base64" -l"ObjectiveSugar" -l"OpenUDID" -l"PureLayout" -l"TDSemiModal" -l"TPKeyboardAvoiding" -l"Toast" -l"WYPopoverController" -l"YLProgressBar" -l"xml2" -framework "CoreData" -framework "CoreGraphics" -framework "CoreText" -framework "Foundation" -framework "ImageIO" -framework "MediaPlayer" -framework "MessageUI" -framework "MobileCoreServices" -framework "QuartzCore" -framework "Security" -framework "SystemConfiguration" -framework "UIKit"

That code simply is meant to remove the '-l"xml2"' from the LDFLAGS for CLEngine, as it is a static library.  We don't want to try to link a dynamic library to it...we will be linking the xml2 dynamic library with the app itself.

I always thought there was a nice way to do that from within the pod's configuration data before it's written to disk, rather than gsub'ing in the output, but I couldn't figure out how to do that.

Make sense?  Is there a better way?

**Comment by segiddins** [comment]
Integrating into static libs / frameworks isn't really supported at the moment, instead we recommend making those pods as well so cocoapods can handle all of the configuration.

**Comment by kcorey** [comment]
Hey Steve, thanks for writing, by the way!

This project was started by university students 5 years back.   There's been constant development since, and now there are almost 1000 classes, some them very complex.

It'll be hard for me to sell the idea of a redesign of the package structure to the powers that be, especially since this worked up until migrating to the latest version of cocoapods.

Any other suggestions?  Links to appropriate RTFMs?

-Ken

**Comment by segiddins** [comment]
It's hard to tell without having a project we can run that reproduces the issue, sorry

**Comment by kcorey** [comment]
Ha!  I was afraid you'd say that.

I'll see if I can make a sample app that demonstrates this.

-Ken

**Comment by kcorey** [comment]
Ah, while building the sample app, I found it.

I'm writing this up hoping that it's helpful to some other newbie such as myself.  This will all be old hat to you, Sam.

The app has multiple targets.  Turns out that the way cocoapods connects the linker flags with the targets to tell them what it should link to is the .xcconfig files.

In my case, "Pods-CLEngine-debug.xcconfig", "Pods-CLEngine-release.xcconfig", "Pods-CLEngine-testing.xcconfig".  

These are connected by clicking on the project name ("CLEngine"), then clicking on the project ("CLEngine"), and then opening up the configurations.  

For each target, you set the 'Based on Configuration File' column to point to a relevant .xcconfig file...AND then you tap on your target, tap on 'Build Settings', and then in 'Other Linker Flags', you must have $(inherited) somewhere in there.

Since I use Reveal during development, I also needed "-all_load -ObjC -lz", so my 'Other Linker Flags'  is: "-all_load -ObjC -lz $(inherited)".

When that's entered, the Other Linker Flags that are shown will be expanded to show all the flags imported from the OTHER_LDFLAGS variable in the .xcconfig file.

As soon as I did that, the target was linked against the correct pod libraries, and the world was good again.
